primitives/account/src/lib.rs

```diff
@@ -20,14 +20,83 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use parity_scale_codec::{Decode, Encode};
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 use sha3::{Digest, Keccak256};
 use sp_core::{ecdsa, H160, H256};
 
 #[cfg(feature = "std")]
 pub use serde::{de::DeserializeOwned, Deserialize, Serialize};
 
+//TODO Maybe this should be upstreamed into Frontier (And renamed accordingly) so that it can
+// be used in palletEVM as well. It may also need more traits such as AsRef, AsMut, etc like
+// AccountId32 has.
+
+/// The account type to be used in Moonbeam. It is a wrapper for 20 fixed bytes. We prefer to use
+/// a dedicated type to prevent using arbitrary 20 byte arrays were AccountIds are expected. With
+/// the introduction of the `scale-info` crate this benefit extends even to non-Rust tools like
+/// Polkadot JS.
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[derive(
+	Eq,
+	PartialEq,
+	Copy,
+	Clone,
+	Encode,
+	Decode,
+	sp_core::RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+	Default,
+	PartialOrd,
+	Ord,
+)]
+pub struct AccountId20(pub [u8; 20]);
```

Comment:
@joelamouche This is the type. The moonbase runtime's metadata contains this. So I guess the path is `account::AccountId20` which makes sense from a Rust perspective.
```json
          {
            id: 0
            type: {
              path: [
                account
                AccountId20
              ]
              params: []
              def: {
                Composite: {
                  fields: [
                    {
                      name: null
                      type: 1
                      typeName: [u8; 20]
                      docs: []
                    }
                  ]
                }
              }
              docs: []
            }
          }
```